service/gcs/bridge/bridge_unit_test.go

```diff
@@ -0,0 +1,619 @@
+package bridge
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"io"
+	"strings"
+	"sync"
+	"testing"
```

Comment:
ginkgo doesn't seem particularly suited for unit testing because each test assumes a very specific set of handlers/errors/fakes/etc. What ends up happening is you have to put a ton of global state in the context's that some tests use and others don't. It seems better to not use ginkgo for the unit tests parts. Do you have a strong objection here?